clearnode/pkg/sign/sign.go

```diff
@@ -35,6 +35,36 @@ type Address interface {
 // Signature is a generic byte slice representing a cryptographic signature.
 type Signature []byte
 
+// Algorithm represents the cryptographic algorithm used for signatures.
+type Algorithm uint8
+
+const (
+	AlgorithmKeccak256 Algorithm = iota
+	AlgorithmECDSA
+	AlgorithmUnknown = 255
+)
+
+// String returns the string representation of the algorithm.
+func (a Algorithm) String() string {
+	switch a {
+	case AlgorithmKeccak256:
+		return "Keccak256"
+	case AlgorithmECDSA:
+		return "ECDSA"
+	default:
+		return "Unknown"
+	}
+}
+
+// Alg returns the algorithm used for this signature based on its length and structure.
+func (s Signature) Alg() Algorithm {
+	if len(s) == 65 {
+		// Standard Ethereum signature format (r: 32 bytes, s: 32 bytes, v: 1 byte)
+		return AlgorithmKeccak256
```

Comment:
That is ECDSA, Keccak256 is a hashing algorithm, it shouldn't be listed here :)

Reply:
But actually I think we will have one algorithm per platform, so either types here should have the names of platforms (aka package names) or package names should be named as algorithms.

I personally don't like the idea of renaming `ethereum` package to `ecdsa`, maybe just `eth` instead. And then the method name here would be either `Type()` or `Platform()` instead of `Alg()`, because each sign/verify depends not only on algorithm (for example not each ECDSA signature has `v` part incremented with 27)